### PR TITLE
feat: implement some "Best Practices" examples

### LINKED ISSUE #95: Implement example: functions_tips_infinite_retries
Line numbers are approximate:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/tips/infinite_retries/infinite_retries.go#L15-L53

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/tips/main.py#L15-L17

### LINKED ISSUE #96: Implement example: functions_tips_lazy_globals
Line numbers are approximate:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/tips/lazy.go#L15-L54

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/tips/main.py#L89-L117

### LINKED ISSUE #97: Implement example: functions_tips_retry
Line numbers are approximate:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/tips/retry.go#L15-L56

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/tips/main.py#L29-L31

### PR DESCRIPTION
**Pull request by coryan**
These examples are used in the "Best Practices" page(s):

https://cloud.google.com/functions/docs/bestpractices/tips
https://cloud.google.com/functions/docs/bestpractices/retries

Fixes #95, fixes #96, and fixes #97

### CODE REVIEW

**Comment by coryan** [comment]
Ping

File: examples/site/tips_infinite_retries/tips_infinite_retries.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// limitations under the License.
+
+// [START functions_tips_infinite_retries]
+#include <google/cloud/functions/cloud_event.h>
+#include <iostream>
+
+namespace gcf = ::google::cloud::functions;
+
+namespace {
+auto constexpr kMaxAge = std::chrono::seconds(10);
```
Comment (devbww) [inline examples/site/tips_infinite_retries/tips_infinite_retries.cc:22]: `#include <chrono>`

File: examples/site/tips_retry/tips_retry.cc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// [START functions_tips_retry]
+#include <google/cloud/functions/cloud_event.h>
+#include <nlohmann/json.hpp>
+#include <iostream>
+
+namespace gcf = ::google::cloud::functions;
+
+namespace {
+auto constexpr kMaxAge = std::chrono::seconds(10);
```
Comment (devbww) [inline examples/site/tips_retry/tips_retry.cc:23]: Unused?

File: examples/site/tips_infinite_retries/tips_infinite_retries.cc (hunk 1 above)
Comment (coryan) [reply examples/site/tips_infinite_retries/tips_infinite_retries.cc:23]: Fixed.

File: examples/site/tips_retry/tips_retry.cc (hunk 2 above)
Comment (coryan) [reply examples/site/tips_retry/tips_retry.cc]: Removed.

**Review by devbww — APPROVED** [review]

**Pull request merged**